libraries/AP_GPS/AP_GPS_GSOF.cpp

```diff
@@ -151,7 +151,7 @@ AP_GPS_GSOF::requestBaud(const uint8_t portindex)
 
     buffer[4] = packetcount++;
 
-    uint8_t checksum = 0;
+    uint8_t checksum;
```

Comment:
Nope, sorry, this is a stack variable and must be zeroed.  Perhaps the style guide could be more clear on the matter?

Similarly elsewhere

Reply:
An example could help, but I just didn't read closely enough.